usagov/usagov-2021
# usagov-2077-usps-api-error: update code to remove specail characters …

…from the address


## Jira Task


https://cm-jira.usa.gov/browse/USAGOV-2077

## Description

Update logic to remove special characters in the address field before submitting to the USPS API
## Type of Changes

  - Add screenshot showing what it should look like

## Testing Instructions



### Change Requirements



### Validation Steps

- Goto http://localhost/elected-officials and input the address below into the proper fields
1 River Place, Apt #2908
New York
New York (NY)
10036
<img width="1512" alt="Screenshot 2024-12-13 at 10 08 34 AM" src="https://github.com/user-attachments/assets/c7af0377-17dd-48d9-9b6f-a214633019b5" />

- After submitting the address you should see the result without the special character in the address
<img width="1512" alt="Screenshot 2024-12-13 at 10 09 29 AM" src="https://github.com/user-attachments/assets/19f79acd-0988-4518-b239-41b2ad0d55da" />

- You SHOULD NOT see the xml error
<img width="1512" alt="Screenshot 2024-12-13 at 10 10 44 AM" src="https://github.com/user-attachments/assets/3306b7a3-6a87-4843-bd81-5d48513efd57" />

- Repeat this verification on the Spanish page http://localhost/es/funcionarios-electos
<img width="1512" alt="Screenshot 2024-12-13 at 10 16 41 AM" src="https://github.com/user-attachments/assets/b117a0e2-3588-4d93-8add-421973af49b0" />

![Screenshot 2024-12-13 at 10 16 45 AM](https://github.com/user-attachments/assets/d686201d-42f6-473c-9682-16fd3b7b7285)

![Screenshot 2024-12-13 at 10 16 48 AM](https://github.com/user-attachments/assets/5b0f41c3-6583-4a1d-a7df-92b89e3ef4fa)

## Security Review




## Reviewer Reminders

- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions

Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
3. Build and deploy the changes.
4. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.
